### PR TITLE
Add Effects and Glue

### PR DESCRIPTION
**Pull request by lukewilliamboswell**
TODO

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1 @@
+platform/glue/**/* linguist-generated=true
```
Comment (lukewilliamboswell) [inline .gitattributes:1]: Using this so the glue files are collapsed by default in PRs.

File: platform/glue.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+# https://vaneyckt.io/posts/safer_bash_scripts_with_set_euxo_pipefail/
+set -euxo pipefail
+
+echo "Removing old glue files..."
+rm -rf platform/glue
+
+echo "Generating glue files..."
+roc glue ../roc/crates/glue/src/RustGlue.roc platform/glue platform/main-glue.roc
+
+echo "NOTE: manually fix any errors in platform/glue files"
```
Comment (lukewilliamboswell) [inline platform/glue.sh:12]: There are issues such as [#6012](https://github.com/roc-lang/roc/issues/6012) which can be manually fixed when re-generating glue.

File: platform/main-glue.roc [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    D InternalFile.ReadErr, 
+    E InternalFile.WriteErr,
+    F InternalTcp.Stream,
+    G InternalTcp.ConnectErr,
+    H InternalTcp.StreamErr,
+    I InternalTcp.ConnectResult,
+    J InternalTcp.WriteResult,
+    K InternalTcp.ReadResult,
+    L InternalTcp.ReadExactlyResult
+]
```
Comment (lukewilliamboswell) [inline platform/main-glue.roc:25]: We can add types here so they will be included in the generated glue. Generation fails when running glue on platform/main.roc.

**Comment by Anton-4** [comment]
@lukewilliamboswell are the included glue files just the result of the glue.sh script without any custom modifications?
If so, could you remove the glue files?

I think it's dangerous to get into the habit of just running code that uses unreviewable large glue files. You are a trustworthy source but your account can be compromised so it's important to avoid blind trust.

**Comment by lukewilliamboswell** [comment]
Superseded by #188

**Pull request closed without merging**